modules/apps/item-selector/item-selector-impl/src/main/java/com/liferay/item/selector/internal/GroupItemSelectorGroupUtil.java

```diff
@@ -0,0 +1,139 @@
+import com.liferay.portal.kernel.service.permission.GroupPermissionUtil;
+import com.liferay.portal.kernel.util.LinkedHashMapBuilder;
+import com.liferay.portal.kernel.util.ListUtil;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+/**
+ * @author István András Dézsi
+ */
+public class GroupItemSelectorGroupUtil {
+
+	public static List<Group> getGroups(
+			long companyId, long[] classNameIds, String keywords,
+			LinkedHashMap<String, Object> groupParams, int start, int end)
+		throws PortalException {
+
+		try {
+			List<Group> groups = GroupLocalServiceUtil.search(
+				companyId, classNameIds, keywords, groupParams, start, end,
+				null);
+
+			return _fillGroupsPage(
+				_filterGroups(groups), groups.size(), classNameIds, companyId,
+				keywords, groupParams, start, end);
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException);
+
+			return Collections.emptyList();
+		}
+	}
+
+	public static int getGroupsCount(
+		long companyId, long[] classNameIds, String keywords) {
+
+		return GroupServiceUtil.searchCount(
+			companyId, classNameIds, keywords,
+			LinkedHashMapBuilder.<String, Object>put(
+				"actionId", ActionKeys.VIEW
+			).put(
+				"site", Boolean.TRUE
+			).build());
+	}
+
+	private static List<Group> _fillGroupsPage(
+			List<Group> groups, int pageSize, long[] classNameIds,
+			long companyId, String keywords,
+			LinkedHashMap<String, Object> groupParams, int start, int end)
+		throws PortalException {
+
+		int delta = end - start;
+
+		int groupsCount = getGroupsCount(companyId, classNameIds, keywords);
+
+		int pageCount =
+			(groupsCount / delta) + (((groupsCount % delta) == 0) ? 0 : 1);
+
+		int pageIndex = (int)Math.ceil((double)start / delta);
+
+		while ((groups.size() < pageSize) && (pageIndex < pageCount)) {
+			pageIndex++;
+
+			start += delta;
+			end += delta;
+
+			List<Group> remainigFilteredGroups = _filterGroups(
+				GroupLocalServiceUtil.search(
+					companyId, classNameIds, keywords, groupParams, start, end,
+					null));
+
+			int difference = pageSize - groups.size();
+
+			groups.addAll(
+				ListUtil.subList(remainigFilteredGroups, 0, difference));
+		}
+
+		return groups;
+	}
+
+	private static List<Group> _filterGroups(List<Group> groups)
+		throws PortalException {
+
+		PermissionChecker permissionChecker =
+			GuestOrUserUtil.getPermissionChecker();
+
+		return ListUtil.filter(
+			groups, group -> _hasViewPermission(permissionChecker, group));
+	}
+
+	private static boolean _hasViewPermission(
+		PermissionChecker permissionChecker, Group group) {
+
+		boolean hasViewPermission = false;
+
+		try {
+			hasViewPermission =
+				group.isCompany() ||
+				permissionChecker.isGroupAdmin(group.getGroupId()) ||
+				GroupPermissionUtil.contains(
+					permissionChecker, group, ActionKeys.VIEW);
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException);
+		}
+
+		return hasViewPermission;
```

Comment:
Can you avoid modifying the variable? Something like:
```
try {
    return group.isCompany() || ...
}
catch (...) {
    ...
    return false;
}
```

Reply:
I have changed it according to your suggestion. Then the source formatter rewrote it to its current form.
See the pushed code.